### PR TITLE
Add smoke test for production build

### PR DESCRIPTION
**Pull request by planger**
#### What it does

Currently we don't seem to have any automated test that verifies whether the production build works as expected.
For a recent issue and its root cause, which remained unrevealed by the CI, see https://github.com/eclipse-theia/theia/issues/12962.

With this PR, we add a new Github workflow, which builds the example browser app in production mode and runs a simple smoke test. This smoke test just opens the app and awaits the loading spinner to disappear and expects the workbench to be visible within a certain timeframe.

Contributed on behalf of STMicroelectronics

#### How to test

The check should be green after https://github.com/eclipse-theia/theia/pull/12964 is merged.

#### Follow-ups

None

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by planger** [comment]
@msujew Sorry for the ping, do you have time to review this change? If not, let me know. Thank you!

File: .github/workflows/production-smoke-test.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: "0 5 * * *" # Runs every day at 5am: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule
```
Comment (msujew) [inline .github/workflows/production-smoke-test.yml:12]: Question: Do we need the schedule, if it's run on every PR and every push to master?

**Review by msujew — APPROVED** [review]
Looks pretty good 👍

File: .github/workflows/production-smoke-test.yml (hunk 1 above)
Comment (planger) [reply .github/workflows/production-smoke-test.yml]: Yeah, you're probably right. I'll remove it.

**Pull request merged**